### PR TITLE
Feature: implement hooks

### PR DESCRIPTION
**Pull request by ECWireless**
Adds `useFarmStats` hook to the rest of the frontend.

- [x] Add stats to `ConfirmDeposit` modal
- [x] Add "view more" dropdown"
- [x] Add stats to deposit page
- [x] Add stats to withdraw page
- [x] Add stats to confirm modals
- [x] Commify stats
- [x] Add stats to stats page
- [x] Add stats to my farms page
- [x] Focus input field
- [x] Make green check clickable
- [x] Add stats link

### CODE REVIEW

**Comment by Evalir** [comment]
Hey guys, what's the state on this? We have huge conflicts now that might need to be solved haha! @ECWireless @crisog

**Comment by ECWireless** [comment]
@Evalir Just waiting for the go ahead from @crisog to mark this as ready for review. I believe he was just adding add a couple new stats to the hook. And conflicts shouldn't be an issue once #17 is merged with this!

**Comment by Evalir** [comment]
Just to clarify—should we merge this into #17 or master?

**Comment by ECWireless** [comment]
> Just to clarify—should we merge this into #17 or master?

Little wonky, but this into `master`, and only after merging #17 into this.

**Comment by Evalir** [comment]
Ah alright!

**Comment by Evalir** [comment]
Also, guys, this is a _huge_ PR... Which commits should I pay attention to?

File: src/hooks/useUserStats.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        const today = dayjs();
+        const parsedEarned = new BigNumber(earned);
+
+        let totalStakeShareSecs = ZERO;
+        let parsedTotalStaked = ZERO;
+        for (let index = 0; index < stakes.length; index++) {
+          const stake: Stake = stakes[index];
+          stake.amount = new BigNumber(stake.amount);
+          parsedTotalStaked = parsedTotalStaked.plus(stake.amount);
+          totalStakeShareSecs = totalStakeShareSecs.plus(stake.amount.times(today.unix() - stake.timestamp));
```
Comment (Evalir) [inline src/hooks/useUserStats.ts:90]: Hmm, this is wrong. We should not mutate React's state ourselves, but rather use the `setStakes` function given by the `useStake` hook. What I'd do is create a new stake array by looping over the existing one, updating the values, and then setting that new array as the new state.

File: src/hooks/useUserStats.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        let averageMultiplier = 0;
+        for (let index = 0; index < stakes.length; index++) {
+          const stake: Stake = stakes[index];
+          const stakeWeight = stake.amount.div(parsedTotalStaked).toNumber();
+
+          const stakeDate = dayjs.unix(stake.timestamp);
+          const secondsUntilMaxBonus = maxBonusDate.diff(stakeDate, 'seconds');
+          const weightedBonusMultiplier = ((secondsUntilMaxBonus * MAX_MULTIPLIER) / bonusPeriodSec) * stakeWeight;
+          averageMultiplier += weightedBonusMultiplier;
+        }
```
Comment (Evalir) [inline src/hooks/useUserStats.ts:108]: Same as above haha

Comment (Evalir) [reply src/hooks/useUserStats.ts:108]: Also haha, noticed we aren't updating stake on this, but I think we could probably write the for loop like

`for (const stake of stakes) {}`

Using newer features. 😄

File: src/hooks/useUserStats.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+        let averageMultiplier = 0;
+        for (let index = 0; index < stakes.length; index++) {
+          const stake: Stake = stakes[index];
+          const stakeWeight = stake.amount.div(parsedTotalStaked).toNumber();
+
+          const stakeDate = dayjs.unix(stake.timestamp);
+          const secondsUntilMaxBonus = maxBonusDate.diff(stakeDate, 'seconds');
+          const weightedBonusMultiplier = ((secondsUntilMaxBonus * MAX_MULTIPLIER) / bonusPeriodSec) * stakeWeight;
+          averageMultiplier += weightedBonusMultiplier;
```
Comment (Evalir) [inline src/hooks/useUserStats.ts:107]: If we're doing this loop for just calculating the whole multiplied, we could probably just do `.reduce()` on the array. 😄

**Review by Evalir — COMMENTED** [review]
Some comments:

File: src/hooks/useUserStats.ts (hunk 1 above)
Comment (amynguy86) [reply src/hooks/useUserStats.ts]: this is so basic, is the intern writing the app :)

**Comment by ECWireless** [comment]
> Also, guys, this is a _huge_ PR... Which commits should I pay attention to?

I believe the most important commits are 18ae4175f254a52a7348009604c3240b588750cf and beyond. The rest are primarily small patches, or commits previously approved PRs. Sorry, should have put small patches on separate branches 😬

**Review by Evalir — APPROVED** [review]
We can finally merge this (After #17, of course)!

**Comment by ECWireless** [comment]
> We can finally merge this (After #17, of course)!

👌 is #17 approved to merge into this?

**Pull request merged**